src/main/java/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineView.java

```diff
@@ -606,8 +607,14 @@ private int triggerBuild(final AbstractProject<?, ?> triggerProject, final Abstr
         if (manualTrigger != null) {
             final Set<String> downstreamProjectsNames =
                     Sets.newHashSet(Splitter.on(",").trimResults().split(manualTrigger.getDownstreamProjectNames()));
-            if (downstreamProjectsNames.contains(project.getFullName())) {
-                configs = manualTrigger.getConfigs();
+            final Iterator<String> downstreamProjectsNamesIter = downstreamProjectsNames.iterator();
+            while (downstreamProjectsNamesIter.hasNext()) {
+                final String downstreamProjectsName = downstreamProjectsNamesIter.next();

```

Comment:
@jglick afair it was used because of performance to not handle all items in object and use just iterator...